src/encoded/static/components/navigation/components/SearchBar.js

```diff
@@ -36,38 +37,57 @@ export class SearchBar extends React.PureComponent {
         return href && typeof href === 'string' && (href.indexOf('/browse/') > -1 || href.indexOf('/search/') > -1);
     }
 
+    static deriveSearchItemTypeFromHref(href){
```

Comment:
Maybe could add in `searchItemType` as parameter here also and return "within" if was previously "within" & on search or browse page... hmm...